internal/buffer/settings.go

```diff
@@ -1,82 +1,86 @@
 package buffer
 
 import (
+	"reflect"
+
 	"github.com/zyedidia/micro/v2/internal/config"
 	"github.com/zyedidia/micro/v2/internal/screen"
 	"golang.org/x/text/encoding/htmlindex"
 )
 
 func (b *Buffer) SetOptionNative(option string, nativeValue interface{}) error {
-	b.Settings[option] = nativeValue
+	if !reflect.DeepEqual(b.Settings[option], nativeValue) {
```

Comment:
And here we have even less reason to not return immediately if they are equal?
i.e. we shouldn't call `b.OptionCallback` if they are equal?

Reply:
Here I fully agree. The large `if`-condition is less readable and the especially regarding `b.OptionCallback`'s documentation:
> // OptionCallback is called after a buffer option value is changed.

There is nothing more to say. :+1: